src/Data/ExistsRE.purs

```diff
@@ -0,0 +1,20 @@
+module Data.ExistsRE where
+
+import Unsafe.Coerce (unsafeCoerce)
+
+-- | A variant of `Exists` that works for type constructors that accept a _row_
+-- | of effects.
+data ExistsRE (f :: # ! -> *)
```

Comment:
I don't know if it makes any semantic difference in PureScript, but is there a way to export this abstractly so that it is not possible to pattern match on it? Otherwise, this looks like an empty type, whence we can prove `False` quite easily.

This is one of the reasons, by the way, I was so strongly in favor of changing PureScript to not export the identities of type synonyms unless you ask it to—always exporting the identity of a type synonym makes data abstraction very difficult, and in many cases impossible.